lexos/views/tokenizer_view.py

```diff
@@ -1,579 +1,197 @@
-                df = pd.concat(
-                    [df, pd.DataFrame(sums, columns=['Total'])], axis=1)
-                df = pd.concat(
-                    [df, pd.DataFrame(averages, columns=['Average'])], axis=1)
-                end_t = timer()
-                elapsed = end_t - start_t
-                print("DataFrame modified.")
-                print(elapsed)
-                # Create the matrix
-                matrix = df.values.tolist()
-                matrix[0][0] = "Terms"
-                end_t = timer()
-                elapsed = end_t - start_t
-                print("DataFrame converted to matrix.")
-                print(elapsed)
-            # Prevent Unicode errors in column headers
-            for i, v in enumerate(matrix[0]):
-                matrix[0][i] = v
-            # Save the column headers and remove them from the matrix
-            # columns = natsorted(matrix[0])
-            columns = matrix[0]
-            if csv_orientation == "filecolumn":
-                columns[0] = "Terms"
-            else:
-                columns[0] = "Documents"
-            del matrix[0]
-            # Prevent Unicode errors in the row headers
-            for i, v in enumerate(matrix):
-                matrix[i][0] = v[0]
-            # Calculate the number of rows in the matrix
-            records_total = len(matrix)
-            # Sort the matrix by column 0
-            matrix = natsorted(matrix, key=itemgetter(0), reverse=False)
-            # Set the table length -- maximum 10 records for initial load
-            if records_total <= 10:
-                end_index = records_total - 1
-                matrix = matrix[0:end_index]
-            else:
-                matrix = matrix[0:9]
-            # escape all the html character in matrix
-            matrix = [[general_functions.html_escape(
-                row[0])] + row[1:] for row in matrix]
-            # escape all the html character in columns
-            columns = [general_functions.html_escape(item) for item in columns]
-            # The first 10 rows are sent to the template as an HTML string.
-            # After the template renders, an ajax request fetches new data
-            # to re-render the table with the correct number of rows.
-            # Create the columns string
-            cols = "<tr>"
-            for s in columns:
-                cols += "<th>" + str(s) + "</th>"
-            cols += "</tr>"
-            # Create the rows string
-            rows = ""
-            for l in matrix:
-                row = "<tr>"
-                for s in l:
-                    row += "<td>" + str(s) + "</td>"
-                row += "</tr>"
-                rows += row
-        # Catch instances where there is no active document (triggers the error
-        # modal)
+
+            # Get the DTM as a list of tuples
+            dtm = get_dtm_matrix(get_session_dtm_options(), file_manager)
```

Comment:
why are we not using matrix model here?

Reply:
My bad. I have not talked about the MVC structure yet. I will try to give the team a quick introduction on it tomorrow.